salt/modules/ansiblegate.py

```diff
@@ -45,7 +45,14 @@
 """
 DEFAULT_TIMEOUT = 1200  # seconds (20 minutes)
 
-__load__ = __non_ansible_functions__ = ["help", "list_", "call", "playbooks"][:]
+__load__ = __non_ansible_functions__ = [
+    "help",
+    "list_",
+    "call",
+    "playbooks",
+    "discover_playbooks",
+    "targets",
+][:]
```

Comment:
What's the point of `[:]`? With or without it, `__load__` and `__non_ansible_functions__` both point to the same object

Reply:
Hmmm, that code was actually introduced by upstream (not sure why exactly), so I simply added new values there and didn't take care of thinking on that.

Now that you mentioned that, it's true that we are actually getting the same object in both `__load__` and `__non_ansible_functions__`, which maybe it is not what was intended when using `[:]`.

In case we do `__load__ = __non_ansible_functions__[:] = ["item1", "item2"]` then we would have different objects, but not sure the reason behind this. Maybe something worth to ask upstream

Reply:
I think this could be simply a way to make both variables not to be the same instance as the static declarared list. I don't know the actual reason of this, but I see this is common in the Salt codebase.

I'll try to get some feedback in the upstream PR https://github.com/saltstack/salt/pull/60056 but I would suggest to not block this PR until getting an answer about it. The current changes on this PR are already fixing the problem we have in SUMA HEAD on the ansible features testing.

If there is any refactor on this area in the future, we can push it later if we consider it. 

Thanks for the review! :+1:

Reply:
Ok, this was clarified by upstream here https://github.com/saltstack/salt/pull/60056#issuecomment-1031457524

I'm pushing the fix for this also in this PR.